compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -804,6 +804,24 @@ abstract class PrepJSInterop extends plugins.PluginComponent
             reporter.error(tree.pos,
                 "@JSBracketAccess methods with two parameters must return Unit")
           }
+
+          val flatParams = sym.paramss.flatten
+
+          for {
+            param <- flatParams
+            if isScalaRepeatedParamType(param.tpe)
+          } {
+            reporter.error(param.pos,
+                "@JSBracketAccess methods may not have repeated parameters")
+          }
+
+          for {
+            firstParam <- flatParams.headOption
+            if firstParam.isParamWithDefault
+          } {
+            reporter.error(firstParam.pos, "The first parameter of a " +
+                "@JSBracketAccess method may not be a default parameter")
```

Comment:
What would it mean for the second parameter to have a default value? If we allow that, we should have a positive test for it. But I believe it should be forbidden.